src/components/MoneyReportHeader.tsx

```diff
@@ -113,11 +113,18 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
     const isArchivedReport = ReportUtils.isArchivedRoomWithID(moneyRequestReport?.reportID);
     const [archiveReason] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${moneyRequestReport?.reportID ?? '-1'}`, {selector: ReportUtils.getArchiveReason});
 
-    const shouldShowPayButton = useMemo(
+    const canIOUBePaid = useMemo(
         () => IOU.canIOUBePaid(moneyRequestReport, chatReport, policy, transaction ? [transaction] : undefined),
         [moneyRequestReport, chatReport, policy, transaction],
     );
 
+    const onlyShowPayElsewhere = useMemo(
+        () => !canIOUBePaid && IOU.canIOUBePaid(moneyRequestReport, chatReport, policy, transaction ? [transaction] : undefined, true),
```

Comment:
> Since we are calling IOU.canIOUBePaid twice

As you can see, I am using these two results (canIOUBePaid and onlyShowPayElsewhere) in different places, so I think we cannot combine these into one.

Reply:
@ZhenjaHorbach Thanks for your suggestion. But if I create a function:

```
    const getCanIOUBePaid  = (onlyShowPayElsewhere)=>IOU.canIOUBePaid(moneyRequestReport, chatReport, policy, transaction ? [transaction] : undefined, onlyShowPayElsewhere)
```

then, to get two variable ```canIOUBePaid``` and ```onlyShowPayElsewhere```, I still need to call the above function two times:

```
    const canIOUBePaid = getCanIOUBePaid(false)

    const onlyShowPayElsewhere = useMemo(
        () => !canIOUBePaid && getCanIOUBePaid(true),
        [iouReport, chatReport, policy, allTransactions, canIOUBePaid],
    );
```

As you can see, it does not resolve your thought:

> Since we are calling IOU.canIOUBePaid twice with the same parameters
Except for the last one